src/v/coproc/pacemaker.cc

```diff
@@ -49,24 +49,51 @@ rpc::backoff_policy wasm_transport_backoff() {
     return rpc::make_exponential_backoff_policy<rpc::clock_type>(1s, 10s);
 }
 
+void pacemaker::save_routes() {
+    (void)ss::with_gate(_gate, [this] {
+        all_routes routes;
+        for (auto& [id, script] : _scripts) {
+            routes.emplace(id, script->get_routes());
+        }
+        return save_offsets(_offs.snap, std::move(routes)).then([this] {
+            if (!_offs.timer.armed()) {
+                _offs.timer.arm(_offs.duration);
+            }
+        });
+    });
+}
+
 pacemaker::pacemaker(unresolved_address addr, sys_refs& rs)
   : _shared_res(
     rpc::reconnect_transport(
       wasm_transport_cfg(addr), wasm_transport_backoff()),
     rs) {
-    _offs.timer.set_callback([] {});
+    _offs.timer.set_callback([this] {
```

Comment:
Oh, it is used there

Reply:
Yeah in a later change it is reintroduced. I hoped to make this easier for reviewers, instead of combining multiple big changes into one

Reply:
I'm open to feedback on how to modify this if anyone has any ideas